File: tests/agent_workloads/adapters/repository_semantic_inspection.py
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: repository_semantic_inspection INPUT_ROOT")

source = Path(sys.argv[1]) / "project" / "examples" / "main.jet"
lines = source.read_text().splitlines()
definitions = sum(line.startswith("fn ") for line in lines)
references = sum(line.lstrip().startswith(("print(", "prepare(")) for line in lines)
print(f"definitions={definitions}")
print(f"references={references}")
print(f"calls={references}")

File: tests/agent_workloads/adapters/repository_semantic_inspection.mjs
import { readFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: repository_semantic_inspection INPUT_ROOT");
}

const source = `${process.argv[2]}/project/examples/main.jet`;
const lines = readFileSync(source, "utf8").split(/\r?\n/);
const definitions = lines.filter((line) => line.startsWith("fn ")).length;
const references = lines.filter((line) => /^\s*(?:print|prepare)\(/.test(line)).length;
console.log(`definitions=${definitions}`);
console.log(`references=${references}`);
console.log(`calls=${references}`);
